### PR TITLE
[needs testing] fix #maintenance_mode? return value

### PR DESCRIPTION
**Pull request by aspiers**
Make `#maintenance_mode?` return a sensible value even when openais is down or uninstalled.  This is especially helpful when it is invoked by Chef's `start_handler`.

### CODE REVIEW

File: chef/cookbooks/crowbar-pacemaker/libraries/maintenance_mode_helpers.rb [hunk 1]
```diff
@@ -19,6 +19,26 @@ module CrowbarPacemaker
...
+      when /Connection refused/
+        # Cluster is not up, so let things proceed so that Chef can
+        # start it if appropriate.
+        return false
+      when /command not found/
+        # pacemaker has been deinstalled?
+        return false
+      end
+
+      if ! $?.success?
```
Comment (tboerger) [inline chef/cookbooks/crowbar-pacemaker/libraries/maintenance_mode_helpers.rb]: ```unless $?.success?```

Comment (aspiers) [inline chef/cookbooks/crowbar-pacemaker/libraries/maintenance_mode_helpers.rb]: I really don't like multi-line `unless` - I find it more effort for the brain to understand.

**Comment by aspiers** [comment]
This change is good but the commit message is wrong - it's not necessary for cluster startup.  I think I got confused in Paris due to lack of sleep.

**Comment by aspiers** [comment]
OK commit message is correct now.

**Comment by vuntz** [comment]
@aspiers: has this been tested? Also, I'm a bit unclear in which case this actually matters, since I don't remember seeing anything wrong because of this.

**Comment by aspiers** [comment]
@vuntz I'll test again.  It's just a cleaner way of handling the corner cases; I don't expect it will be hit except in rare circumstances.  But I guess it might also result in cleaner logs.

**Comment by rsalevsky** [comment]
@aspiers, this repository got merged into [crowbar-ha](https://github.com/crowbar/crowbar-ha).
Please open a new pull request there. __Do NOT merge here anymore!__

**Comment by aspiers** [comment]
Superceded by https://github.com/crowbar/crowbar-ha/pull/13
